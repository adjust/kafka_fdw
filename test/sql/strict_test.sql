CREATE FOREIGN TABLE kafka_strict (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    some_int int,
    some_text text,
    some_date date,
    some_time timestamp
)
SERVER kafka_server OPTIONS
    (format 'csv', topic 'contrib_regress_junk', batch_size '30', buffer_delay '100', strict 'true');

SELECT * FROM kafka_strict WHERE offs>=0 and part=0;

CREATE FOREIGN TABLE kafka_ignore_junk (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    some_int int,
    some_text text,
    some_date date,
    some_time timestamp
)
SERVER kafka_server OPTIONS
    (format 'csv', topic 'contrib_regress_junk', batch_size '30', buffer_delay '100', ignore_junk 'true');

SELECT * FROM kafka_ignore_junk WHERE offs>=0 and part=0;

