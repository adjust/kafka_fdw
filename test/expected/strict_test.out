CREATE FOREIGN TABLE kafka_strict (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    some_int int,
    some_text text,
    some_date date,
    some_time timestamp
)
SERVER kafka_server OPTIONS
    (format 'csv', topic 'contrib_regress_junk', batch_size '30', buffer_delay '100', strict 'true');
SELECT * FROM kafka_strict WHERE offs>=0 and part=0;
ERROR:  extra data after last expected column
CREATE FOREIGN TABLE kafka_ignore_junk (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    some_int int,
    some_text text,
    some_date date,
    some_time timestamp
)
SERVER kafka_server OPTIONS
    (format 'csv', topic 'contrib_regress_junk', batch_size '30', buffer_delay '100', ignore_junk 'true');
SELECT * FROM kafka_ignore_junk WHERE offs>=0 and part=0;
 part | offs | some_int |                  some_text                  | some_date  |        some_time         
------+------+----------+---------------------------------------------+------------+--------------------------
    0 |    0 |       91 | correct line                                | 01-01-2015 | Thu Jan 01 01:31:00 2015
    0 |    1 |      131 | additional data                             | 01-01-2015 | Thu Jan 01 02:11:00 2015
    0 |    2 |      161 | additional data although null               | 01-01-2015 | Thu Jan 01 02:41:00 2015
    0 |    3 |      301 | correct line although last line null        | 01-01-2015 | 
    0 |    4 |      371 | invalidat date                              |            | Thu Jan 01 06:11:00 2015
    0 |    5 |      401 | unterminated string                         |            | 
    0 |    6 |      421 | correct line                                | 01-01-2015 | Thu Jan 01 07:01:00 2015
    0 |    7 |          | invalid number                              | 01-01-2015 | Thu Jan 01 07:11:00 2015
    0 |    8 |      521 | correct line                                | 01-01-2015 | Thu Jan 01 08:41:00 2015
    0 |    9 |          | invalid number, invalid date and extra data |            | Thu Jan 01 09:31:00 2015
    0 |   10 |          |                                             |            | 
(11 rows)

